src/overlays/actors/ovl_En_Sekihi/z_en_sekihi.c

```diff
@@ -32,20 +36,147 @@ const ActorInit En_Sekihi_InitVars = {
+        this->opaDList = sOpaDLists[params];
+        this->xluDList = sXluDLists[params];
+        this->dyna.actor.textId = sTextIds[params];
+        this->dyna.actor.focus.pos.y = this->dyna.actor.world.pos.y + 60.0f;
+        Actor_SetScale(&this->dyna.actor, 0.1f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/func_80A45130.s")
+void EnSekihi_Destroy(Actor* thisx, PlayState* play) {
+    EnSekihi* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/EnSekihi_Update.s")
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/func_80A45164.s")
+void func_80A44DE8(EnSekihi* this, PlayState* play) {
+    CollisionHeader* colHeader = NULL;
+    s32 params;
+    CollisionHeader* sColHeader[] = { &gSekihilColHeader, &gSekihigColHeader, &gSekihinColHeader, &gSekihizColHeader,
+                                      &gObjectZogColHeader };
+
+    params = ENSIKIHI_GET_F(&this->dyna.actor);
+    if (Object_IsLoaded(&play->objectCtx, this->objectIndex)) {
+        this->dyna.actor.objBankIndex = this->objectIndex;
+        this->dyna.actor.draw = EnSekihi_Draw;
+        if (params == 4) {
+            this->actionFunc = func_80A450B0;
+        } else {
+            this->actionFunc = func_80A45130;
+        }
+
+        Actor_SetObjectDependency(play, &this->dyna.actor);
+        DynaPolyActor_Init(&this->dyna, 0);
+        if (sColHeader[params] != 0) {
+            CollisionHeader_GetVirtual(sColHeader[params], &colHeader);
+        }
+
+        this->dyna.bgId = DynaPoly_SetBgActor(play, &play->colCtx.dyna, &this->dyna.actor, colHeader);
+        if ((params == 4) && (INV_CONTENT(ITEM_MASK_ZORA) != ITEM_MASK_ZORA)) {
+            Actor_MarkForDeath(&this->dyna.actor);
+        }
+    }
+}
+
+void func_80A44F40(EnSekihi* this, PlayState* play) {
+    switch (Message_GetState(&play->msgCtx)) {
+        case 4:
+            if ((Message_ShouldAdvance(play)) && (play->msgCtx.currentTextId == 0x1019)) {
+                switch (play->msgCtx.choiceIndex) {
+                    case 0:
+                        func_8019F208();
+                        func_80151938(play, 0x101A);
+                        break;
+                    case 1:
+                        func_8019F208();
+                        func_80151938(play, 0x101B);
+                        break;
+                    case 2:
+                        func_8019F230();
+                        func_801477B4(play);
+                        this->actionFunc = func_80A450B0;
+                        break;
+                }
+            } else {
+                break;
+            }
+            break;
+        case 5:
+            if (Message_ShouldAdvance(play)) {
+                switch (play->msgCtx.currentTextId) {
+                    case 0x1018:
+                        func_80151938(play, (play->msgCtx.currentTextId + 1) & 0xFFFF);
```

Comment:
```suggestion
                        func_80151938(play, play->msgCtx.currentTextId + 1);
```
Is the `& 0xFFFF` needed?